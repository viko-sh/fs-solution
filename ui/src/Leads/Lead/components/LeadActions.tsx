import React from 'react';
import { ActionButton, Button } from '../../../shared/styles';
import { ActionsWrapper, Price } from '../style';
import { LeadProps } from '../type';

export const LeadActions = ({
  price,
  onAccept,
  onDecline
}: Partial<LeadProps>) => {
  return (
    <ActionsWrapper>
      <ActionButton className="accept" onClick={onAccept}>
        Accept
      </ActionButton>
      <Button className="decline" onClick={onDecline}>
        Decline
      </Button>
      <Price>
        <span>${Number(price).toFixed(2)}</span> Lead Invitation
      </Price>
    </ActionsWrapper>
  );
};
