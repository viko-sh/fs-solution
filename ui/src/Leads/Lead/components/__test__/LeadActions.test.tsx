import React from 'react';
import { mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

import { LeadActions } from '../LeadActions';

const defaultProps = {
  id: 7,
  contactName: 'Darth',
  price: 30,
  description: 'Praesent elit ',
  categoryName: 'Electrical',
  suburbName: 'Bondi',
  postCode: '2026',
  createdAt: '2019-11-21T18:57:43.000Z'
};

const mountComponent = (customProps: any = {}) => {
  const props = { ...defaultProps, ...customProps };
  const wrapper = mount(<LeadActions {...props} />);
  return { props, wrapper };
};

describe('LeadActions component', () => {
  it('Should render header', done => {
    const { wrapper } = mountComponent();
    expect(wrapper).toHaveLength(1);
    done();
  });
  it('should be match with snapshot', done => {
    // Arrange (no Act required)
    const { wrapper } = mountComponent();
    // Assert
    expect(wrapper).toMatchSnapshot();
    done();
  });

  it('onAccept should be called on a click', () => {
    const fn = jest.fn();

    const { wrapper } = mountComponent({
      onAccept: fn
    });

    wrapper.find('button.accept').simulate('click');
    expect(fn).toBeCalled();
  });

  it('onDecline should be called on a click', () => {
    const fn = jest.fn();

    const { wrapper } = mountComponent({
      onDecline: fn
    });

    wrapper.find('button.decline').simulate('click');
    expect(fn).toBeCalled();
  });
});
