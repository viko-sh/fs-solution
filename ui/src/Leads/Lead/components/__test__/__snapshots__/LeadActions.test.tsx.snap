// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeadActions component should be match with snapshot 1`] = `ReactWrapper {}`;
